### ISSUE TITLE
PocoDataMySQL CMake configuration file fails to find MySQL client library

### ISSUE THREAD

**Issue by fishnet37222** [pocoproject/poco#3028]
When using the following CMakeLists.txt file, CMake gives the error shown below.

CMakeLists.txt:
```
cmake_minimum_required(VERSION 3.0.0)
project(sqltest VERSION 0.1.0)

set(CMAKE_CXX_STANDARD 17)

find_package(Poco REQUIRED DataMySQL)
include_directories(${Poco_INCLUDE_DIRECTORIES})

add_executable(sqltest main.cpp)
target_link_libraries(sqltest ${Poco_LIBRARIES})
```

CMake error message:
```
[cmake] CMake Error at CMakeLists.txt:9 (add_executable):
[cmake]   Target "sqltest" links to target "MySQL::client" but the target was not
[cmake]   found.  Perhaps a find_package() call is missing for an IMPORTED target, or
[cmake]   an ALIAS target is missing?
```
I am able to work around the issue by modifying the PocoDataMySQLTargets.cmake file so the block starting at line 53 appears as follows:
```
# Create imported target Poco::DataMySQL
find_package(PkgConfig QUIET REQUIRED)
pkg_check_modules(MYSQL REQUIRED QUIET mysqlclient)
add_library(Poco::DataMySQL STATIC IMPORTED)

set_target_properties(Poco::DataMySQL PROPERTIES
  INTERFACE_COMPILE_DEFINITIONS "THREADSAFE;NO_TCL"
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include;${_IMPORT_PREFIX}/include;${MYSQL_INCLUDE_DIRS}"
  INTERFACE_LINK_LIBRARIES "Poco::Data;${MYSQL_LIBRARIES}"
)
```

**Comment by Bjoe** [comment]
In [PocoDataMySQLConfig.cmake](https://github.com/pocoproject/poco/blob/devel/Data/MySQL/cmake/PocoDataMySQLConfig.cmake) it should be add:
```
find_dependency(MySQL)
```
But cmake doesn't provide a `FindMySQL.cmake`, so poco should installd it. But what happens when there is a `FindMySQL.cmake` installed from the MySQL project ....

**Comment by LockDragonBorn** [comment]
find your Poco sources dir,it has a dir named cmake,copy FindMySQL.cmake to /usr/share/cmake-<your version>/Modules。
then find_package(MySQL REQUIRED) before find_package(Poco).

**Comment by matejk** [comment]
Resolved in 1.14.
